### PR TITLE
fix: add admin only endpoint to mark "trusted"

### PR DESCRIPTION
**Pull request by DekusDenial**
## Context

To provide admins access to mark certain template & command as trusted

## Objective

To add separate endpoints for template & command for marking trusted

## References

https://github.com/screwdriver-cd/screwdriver/issues/1427

blocked by: 
https://github.com/screwdriver-cd/data-schema/pull/340

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by DekusDenial** [comment]
test failed due to depending PR https://github.com/screwdriver-cd/data-schema/pull/340

**Comment by tkyi** [comment]
@jithin1987 see https://github.com/screwdriver-cd/datastore-sequelize/pull/52

File: plugins/commands/updateTrusted.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+const boom = require('boom');
+const joi = require('joi');
+const schema = require('screwdriver-data-schema');
+const baseSchema = schema.models.command.base;
+
+module.exports = () => ({
+    method: 'PUT',
+    path: '/commands/{namespace}/{name}/trusted',
+    config: {
+        description: "Update a command's trusted property",
```
Comment (tkyi) [inline plugins/commands/updateTrusted.js]: single quotes?

Comment (DekusDenial) [reply plugins/commands/updateTrusted.js]: eslint not complaining so i guess it’s ok with single q inside double q? unless u want me to backlash escape it?

**Comment by DekusDenial** [comment]
@jithin1987 as per our discussion yesterday, once a template/command got marked trusted it remains that for life unless we explicitly remove that certification. The GET query is basically a group by query after all, as addressed by the other PR

File: plugins/commands/updateTrusted.js (hunk 1 above)
Comment (tkyi) [reply plugins/commands/updateTrusted.js]: I don't mind but just thought it looked odd haha
So used to eslint complaining.

File: plugins/commands/updateTrusted.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            const { trusted } = request.payload;
+
+            const commands = await commandFactory.list({ params: { namespace, name } });
+
+            if (commands.length === 0) {
+                throw boom.notFound(`Command ${namespace}/${name} does not exist`);
+            }
+
+            return Promise.all(
+                commands.map((command) => {
```
Comment (tkyi) [inline plugins/commands/updateTrusted.js]: If I'm reading this correctly, looks like we plan to get **all** the commands with the name/namespace and we're going to go and set each command with the `trusted` value?
If we're checking the MAX value with datastore-sequelize, I think we can just set one of the command versions, right?

File: plugins/commands/updateTrusted.js (hunk 1 above)
Comment (DekusDenial) [reply plugins/commands/updateTrusted.js]: haha, u should try coding in java, again, if never…

File: plugins/commands/updateTrusted.js (hunk 2 above)
Comment (DekusDenial) [reply plugins/commands/updateTrusted.js]: but that mean u will need to find that one version when u gonna remove it which i rarely think that would ever happen once something got trusted

**Comment by DekusDenial** [comment]
@tkyi not really famililar with the admin scope, but is `admin` not enough? The code seems like if you’re a sys admin, you’re also sd admin

File: plugins/commands/updateTrusted.js (hunk 2 above)
Comment (tkyi) [reply plugins/commands/updateTrusted.js]: Well after calling this endpoint, any newly published command versions will not have `trusted: true`, anyways, right?

And for removing it, are we trying to use the same endpoint? Or will we make a separate one? Or just rely on a DB call?

I think it should be pretty easy to find the one version. We never delete commands/templates. You can pass in another param to `factory.list` to find it or construct a pretty simple SQL command I think.

**Comment by tkyi** [comment]
I think the `scope` is for tokens. But I don't think we have `admin` scope in any tokens atm 😞 
That's another thing we need to fix

**Comment by tkyi** [comment]
Okay actually we do push the `admin` scope: https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/buildClusters/create.js#L44-L52
Shouldn't need those other checks then

File: plugins/commands/updateTrusted.js (hunk 2 above)
Comment (DekusDenial) [reply plugins/commands/updateTrusted.js]: ok, since u spotted my laziness, i guess i surrender….

**Comment by DekusDenial** [comment]
just to clarify, SD admin will use these endpoints for (un)marking templates & commands. Even though they are shown in swagger doc but won't matter since they are protected by scope `admin`

File: plugins/templates/updateTrusted.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                security: [{ token: [] }]
+            }
+        },
+        handler: async (request, reply) => {
+            const { name } = request.params;
+            const { templateFactory } = request.server.app;
+            const { trusted } = request.payload;
+
+            // get the earliest entry
+            const templates = await templateFactory.list({
```
Comment (tkyi) [inline plugins/templates/updateTrusted.js]: not sure if this is any better (probably not): https://github.com/screwdriver-cd/models/blob/master/lib/templateFactory.js#L222

Comment (DekusDenial) [reply plugins/templates/updateTrusted.js]: i think the sole purpose here is to get the earliest one, dont really need a flexibility like that function there right?

**Pull request merged**